### PR TITLE
(dsl): Support 'bucketSortPipeline' aggregation

### PR DESCRIPTION
**Pull request by dbulaja98**
Resolves bucketSortPipelineAgg in #118

### CODE REVIEW

**Review by markaya — CHANGES_REQUESTED** [review]

File: modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala [hunk 1]
```diff
@@ -237,6 +292,76 @@ object ElasticAggregationSpec extends ZIOSpecDefault {
         }
       ),
       suite("encoding as JSON")(
+        test("bucket sort") {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala:295]: Maybe to use `bucketSort`.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala [hunk 2]
```diff
@@ -14,6 +14,61 @@ object ElasticAggregationSpec extends ZIOSpecDefault {
   def spec: Spec[TestEnvironment, Any] =
     suite("ElasticAggregation")(
       suite("constructing")(
+        test("bucket sort") {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala:17]: Maybe use `bucketSort`.

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 3]
```diff
@@ -32,6 +33,61 @@ sealed trait ElasticAggregation { self =>
 
 sealed trait SingleElasticAggregation extends ElasticAggregation
 
+sealed trait BucketSortAggregation extends SingleElasticAggregation with HasSize[BucketSortAggregation] with WithAgg {
+
+  /**
+   * Sets the starting offset from where the [[zio.elasticsearch.aggregation.BucketSortAggregation]] return results.
```
Comment (arnoldlacko) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:39]: Strange sentence, can we rephrase it?

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 4]
```diff
@@ -125,6 +167,9 @@ private[elasticsearch] final case class Terms(
   def withSubAgg(aggregation: SingleElasticAggregation): TermsAggregation =
     self.copy(subAggregations = aggregation +: subAggregations)
 
+  private[elasticsearch] def paramsToJson: Json =
+    Obj(name -> paramsToJsonHelper)
```
Comment (arnoldlacko) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:171]: This is a bit strange as it looks like a "helper" is what is being included in the json. Would the following renaming make sense?
- paramsToJson -> toJson
- paramsToJsonHelper -> jsonParams

File: modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala [hunk 5]
```diff
@@ -26,6 +26,73 @@ sealed trait Sort {
   def paramsToJson: Json
 }
 
+object Sort {
+
+  /**
+   * Constructs an instance of [[SortByFieldOptions]] to sort by `_count` field in the context of an aggregation.
```
Comment (arnoldlacko) [inline modules/library/src/main/scala/zio/elasticsearch/query/sort/Sort.scala:32]: Can any `Sort` be used in the context of aggregations or only these newly added ones?
Should we have an abstraction/restriction to distinguish sorting parameters for aggregation buckets?

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala (hunk 3 above)
Comment (dbulaja98) [reply modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:39]: Aggred to replace `return` with `returns`.

**Review by drmarjanovic — APPROVED** [review]

**Review by markaya — APPROVED** [review]

**Pull request merged**